flaml/autogen/code_utils.py

```diff
@@ -1,56 +1,260 @@
 import signal
 import subprocess
 import sys
+import os
 from typing import List, Dict, Tuple, Optional, Union, Callable
-from flaml import oai
+import re
+import time
+from flaml.autogen import oai, DEFAULT_MODEL, FAST_MODEL
+
+# Regular expression for finding a code block
+CODE_BLOCK_PATTERN = r"```python\n(.*?)\n```"
```

Comment:
Is the code always in python? What if the markdown doesn't contain the keyword "python" in the code block? 

How about change "python" to "\w*"? 

r"```\w*\n(.*?)\n```"